runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/fn/control/BeamFnMapTaskExecutorTest.java

```diff
@@ -423,6 +434,187 @@ public void close() {}
+            .putPtransforms(GRPC_READ_ID, FAKE_ELEMENT_COUNT_METRICS)
+            .putPtransforms(
+                stepName,
+                BeamFnApi.Metrics.PTransform.newBuilder()
+                    .addUser(
+                        BeamFnApi.Metrics.User.newBuilder()
+                            .setMetricName(metricName)
+                            .setCounterData(
+                                BeamFnApi.Metrics.User.CounterData.newBuilder()
+                                    .setValue(finalCounterValue)))
+                    .build())
+            .build();
+
+    final int expectedCounterValue = 5;
+    final BeamFnApi.MonitoringInfo expectedMonitoringInfo =
+        BeamFnApi.MonitoringInfo.newBuilder()
+            .setUrn("beam:metric:user:ExpectedCounter")
+            .setType("beam:metrics:sum_int_64")
+            .putLabels("PTRANSFORM", "ExpectedPTransform")
+            .setMetric(
+                BeamFnApi.Metric.newBuilder()
+                    .setCounterData(
+                        BeamFnApi.CounterData.newBuilder()
+                            .setInt64Value(expectedCounterValue)
+                            .build())
+                    .build())
+            .build();
+
+    InstructionRequestHandler instructionRequestHandler =
+        new InstructionRequestHandler() {
+          @Override
+          public CompletionStage<InstructionResponse> handle(InstructionRequest request) {
+            switch (request.getRequestCase()) {
+              case REGISTER:
+                return CompletableFuture.completedFuture(responseFor(request).build());
+              case PROCESS_BUNDLE:
+                return MoreFutures.supplyAsync(
+                    () -> {
+                      processBundleLatch.await();
+                      return responseFor(request)
+                          .setProcessBundle(
+                              BeamFnApi.ProcessBundleResponse.newBuilder()
+                                  .setMetrics(deprecatedMetrics)
+                                  .addMonitoringInfos(expectedMonitoringInfo))
+                          .build();
+                    });
+              case PROCESS_BUNDLE_PROGRESS:
+                progressSentLatch.countDown();
+                return CompletableFuture.completedFuture(
+                    responseFor(request)
+                        .setProcessBundleProgress(
+                            BeamFnApi.ProcessBundleProgressResponse.newBuilder()
+                                .setMetrics(deprecatedMetrics)
+                                .addMonitoringInfos(expectedMonitoringInfo))
+                        .build());
+              default:
+                throw new RuntimeException("Reached unexpected code path");
+            }
+          }
+
+          @Override
+          public void close() {}
+        };
+
+    Map<String, DataflowStepContext> stepContextMap = new HashMap<>();
+    NameContext nc =
+        new NameContext() {
```

Comment:
Do you need to override this and make a new class? Or is it possible to just instantiate an existing class and passing in these strings?